gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/publisher/CopyDataPublisher.java

```diff
@@ -185,27 +191,35 @@ private static Multimap<CopyEntity.DatasetAndPartition, WorkUnitState> groupByFi
    * and versionStrategy (usually relevant to modTime as well), since they are subject to change with Publish(rename)
    */
   private void preserveFileAttrInPublisher(CopyableFile copyableFile) throws IOException {
-    // Preserving File ModTime, and set the access time to an initializing value when ModTime is declared to be preserved.
-    if (copyableFile.getPreserve().preserve(PreserveAttributes.Option.MOD_TIME)) {
-      fs.setTimes(copyableFile.getDestination(), copyableFile.getOriginTimestamp(), -1);
+    if (!copyableFile.getFileStatus().isFile() && state.getProp(DatasetUtils.DATASET_PROFILE_CLASS_KEY, "").equals(
+        ManifestBasedDatasetFinder.class.getCanonicalName())){
+      FileStatus dstFile = this.fs.getFileStatus(copyableFile.getDestination());
+      // User specifically try to copy dir metadata (which should be only doable in manifest copy), so we change the group and permissions on destination even the dir already existed
+      FileAwareInputStreamDataWriter.safeSetPathPermission(this.fs, copyableFile.getDestination(),
+          FileAwareInputStreamDataWriter.addExecutePermissionsIfRequired(dstFile, copyableFile.getDestinationOwnerAndPermission()));
```

Comment:
since it's not called anywhere else outside the class (given it was formerly `private`), let's perform the `addExecutePermissionsIfRequired` call within.  since it seems we want it to happen all the time, let's not bring the potential for the caller to forget